tools/download-test.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+while [[ $# -gt 1 ]]
+do
+  key="$1"
+
+  case "$key" in
+      -t|--tap)
+      TAP="$2"
+      shift # past argument
+      ;;
+      *) # unknown option
+      ;;
+  esac
+  shift # past argument or value
+done
+
+if [ "$DOWNLOAD_LOCATION" == "Nightly" ]; then
+	DOWNLOAD_DIR="nightly"
+else
+	DOWNLOAD_DIR="release"
+fi
+
+LINK=`rsync rsync://unencrypted.nodejs.org/nodejs/$DOWNLOAD_DIR/ | grep $NODE_VERSION | sort -t. -k 1,1n -k 2,2n -k 3,3n | tail -1 | awk '{print $5}'`
+
+if [ -z "$NODE_VERSION" ]; then
+  echo "NODE_VERSION is undefined! Please declare NODE_VERSION"
+  exit 1
+fi
+
+# Remove old files
+[ "$TAP" ] && rm -rf "$TAP"
+rm -rf v*
+
+# Scrape the download server
+rsync -av rsync://unencrypted.nodejs.org/nodejs/"$DOWNLOAD_DIR"/"$LINK" .
+
+num=1
+cat "$LINK"/SHASUMS256.txt | { while read line
+do
```

Comment:
Maybe just me, but I find this layout really confusing, could you do (change indentation):

```bash
testNumber=1
cat "$LINK"/SHASUMS256.txt | {
  while read line; do
    sha=$(echo "${line}" | awk '{print $1}')
    ...
  done
  let testNumber--
  [ "$tapFile" ] && echo "1..$testNumber" >> "$TAP"
}
```